lessons/beginners/basic-functions/index.md

```diff
@@ -0,0 +1,213 @@
+print(2, 3, 4, sep=', ')
+```
+
+Základní funkce na načtení vstupu, `input`,
+vypíše otázku, počká na text od uživatele a ten vrátí jako řetězec.
+
+```python
+input('zadej vstup: ')
+```
+
+Kontrolní otázky:
+
+* Je `input` „normální“ funkce, nebo procedura?
+* Co bere funkce `input` jako argument?
+* Jaká je návratová hodnota funkce `input`?
+
+{% filter solution %}
+Funkce `input` vrací hodnotu, se kterou může program dál pracovat.
+Zařadil bych ji tedy mezi „normální“ funkce.
+
+Jako argument bere `input` otázku, na kterou se uživatele zeptá.
+
+Návratová hodnota funkce `input` je řetězec s odpovědí uživatele.
+{% endfilter %}
+
+
+
+## Převádění typů
+
+
+Co ale když nechceme pracovat s řetězcem, ale třeba s číslem?
+Tady nám pomůže skupina funkcí, které umí převádět čísla na řetězce a zpátky.
+Každý ze tří <em>typů</em> (angl. <em>types</em>) proměnných, které zatím známe,
+má funkci, která vezme nějakou hodnotu a vrátí podobnou hodnotu „svého“ typu.
+Na celá čísla je funkce `int` (z angl. *integer*), na reálná čísla je `float`
+(z angl. *floating-point*), a pro řetězce `str` (z angl. *string*).
+
+```python
+int(x)              # převod na celé číslo
+float(x)            # převod na reálné číslo
+str(x)              # převod na řetězec
+```
+
+Příklady:
+
+```python
+3 == int('3') == int(3.0) == int(3.141) == int(3)
+8.12 == float('8.12') == float(8.12)
+8.0 == float(8) == float('8') == float(8.0)
+'3' == str(3) == str('3')
+'3.141' == str(3.141) == str('3.141')
+```
+Ne všechny převody jsou možné:
+
+```python
+int('blablabla')    # chyba!
+float('blablabla')  # chyba!
+int('8.9')          # chyba!
+```
+
+…a jak si poradit s chybou, která nastane,
+když použiješ špatnou hodnotu, si řekneme později.
+
+### Převádění a `input`
+
+Převádění typů se často používá při načítání vstupu, třeba jako:
+
+```python
+cislo = int(input('Zadej číslo: '))
+```
+
+Jak Python vyhodnotí tento výraz?
+Zadá-li uživatel <kbd>4</kbd><kbd>2</kbd><kbd>Enter</kbd>,
+funkce `input` vrátí řetězec`'42'`.
+Ten pak funkce `int` vezme jako argument, udělá podle něj číslo a to
+číslo vrátí:
+
+```python
+cislo = int(input('Zadej číslo: '))
+      #     ╰─────────┬─────────╯
+cislo = int(        '42'          )
+      # ╰────────────┬────────────╯
+cislo =             42
+```
+
+
+## Matematické funkce
+
+Matematika je občas potřeba, takže se pojďme
+podívat, jak v Pythonu pracovat s čísly.
+
+Jedna zajímavá matematická funkce je k dispozici vždy:
+
+```python
+round(cislo)    # zaokrouhlení
+```
+
+Spousta dalších není k dispozici od začátku programu.
+Ne každý má rád matematiku, a ne ve všech druzích programu jsou takové operace
+potřeba.
+Proto musíš předem – typicky na začátku souboru – říct, že je budeš používat.
+To se dělá *naimportováním* z modulu `math`:
+
+```python
+from math import sin, cos, tan, sqrt, floor, ceil
+```
+
+Naimportované funkce pak můžeš použít:
+
+```python
+sin(uhel)       # sinus
+cos(uhel)       # kosinus
+tan(uhel)       # tangens
```

Comment:
Poznamenal bych tu, že jde o úhel v radianech, nikoliv ve stupních.
Případně i jak to převést.

Reply:
Poznamenáno. Přijde mi toho tam příliš, takže časem bych to radši odstranil a nechal jen na taháku, ale předtím budu muset kouknout co na to navazuje.